### DISCUSSION TITLE
Starting Pocketbase programmatically in a custom build

### DISCUSSION THREAD

**Discussion by proxima-b-alpha** [pocketbase/pocketbase#2100] [Q&A] [answered]
According to the [Pocketbase doc](https://pocketbase.io/docs/use-as-framework/). We can use PocketBase as a framework and do a custom build.

**Example:**

```
import (
    "log"
    "github.com/pocketbase/pocketbase"
)

func main() {
    app := pocketbase.New()

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

When we do `go build` for above code it creates executable which we can run as:

`./custom-build.exe serve`

Instead of starting the application from outside using `serve` command is it possible to start it from the code (programmatically)? So that we can just run the custom build without any commands `./custom-build.exe` and have the application running.

I looked into PocketBase function but did not find a way to programmatically start the application.

**Example:**

`Start()` - Does not take any commands or flags.

`NewWithConfig()` - Can only set custom config but cannot start the application

```
app := pocketbase.NewWithConfig(&pocketbase.Config{
    DefaultDataDir: "",
    DefaultDebug:   false,
})
```

**Comment by ganigeorgiev** [comment, marked as answer]
You can check https://github.com/pocketbase/pocketbase/discussions/1684 and https://github.com/pocketbase/pocketbase/discussions/1953#discussioncomment-5205828.
